tests/commonTemplates_test.go

```diff
@@ -122,4 +128,86 @@ var _ = Describe("Common templates", func() {
+			baseRequirement, err := labels.NewRequirement("template.kubevirt.io/type", selection.Equals, []string{"base"})
+			Expect(err).To(BeNil())
+
+			var versionRequirement *labels.Requirement
+			versionRequirement, err = labels.NewRequirement("template.kubevirt.io/version", selection.NotEquals, []string{currentVersion})
+			Expect(err).To(BeNil())
+
+			labelsSelector := labels.NewSelector().Add(*baseRequirement, *versionRequirement)
+
+			opts := client.ListOptions{
+				LabelSelector: labelsSelector,
+			}
+
+			var oldTemplates templatev1.TemplateList
+
+			err = apiClient.List(ctx, &oldTemplates, &opts)
+			Expect(err).To(BeNil())
+
+			for _, template := range oldTemplates.Items {
+				if value, ok := template.Labels["template.kubevirt.io/version"]; ok && value == oldVersion {
+					Expect(template.Labels["os.template.kubevirt.io/some-os"]).To(Equal(""), "os.template.kubevirt.io should be empty")
+					Expect(template.Labels["flavor.template.kubevirt.io/test"]).To(Equal(""), "flavor.template.kubevirt.io should be empty")
+					Expect(template.Labels["workload.template.kubevirt.io/server"]).To(Equal(""), "workload.template.kubevirt.io should be empty")
+					Expect(template.Labels["template.kubevirt.io/type"]).To(Equal("base"), "template.kubevirt.io/type should equal base")
+					Expect(template.Labels["template.kubevirt.io/version"]).To(Equal(oldVersion), "template.kubevirt.io/version should equal "+oldVersion)
+
+				}
+				if value, ok := template.Labels["template.kubevirt.io/version"]; ok && value == currentVersion {
+					Expect(template.Labels["os.template.kubevirt.io/some-os"]).To(Equal("true"), "os.template.kubevirt.io should not be empty")
+					Expect(template.Labels["flavor.template.kubevirt.io/test"]).To(Equal("true"), "flavor.template.kubevirt.io should not be empty")
+					Expect(template.Labels["workload.template.kubevirt.io/server"]).To(Equal("true"), "workload.template.kubevirt.io should not be empty")
+					Expect(template.Labels["template.kubevirt.io/type"]).To(Equal("base"), "template.kubevirt.io/type should equal base")
+					Expect(template.Labels["template.kubevirt.io/version"]).To(Equal(currentVersion), "template.kubevirt.io/version should equal "+currentVersion)
+				}
+
+			}
+		})
+
+	})
 })
+
+func getTemplateWithCurrentVersion(namespace, version string) *templatev1.Template {
+	return &templatev1.Template{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Template",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "ok-test-template",
+			Namespace: namespace,
+			Labels: map[string]string{
+				"os.template.kubevirt.io/some-os":      "true",
+				"flavor.template.kubevirt.io/test":     "true",
+				"template.kubevirt.io/type":            "base",
+				"workload.template.kubevirt.io/server": "true",
+				"template.kubevirt.io/version":         version,
+			},
+		},
+	}
+}
+
+func getTemplateWithOldVersion(indexStr, namespace, version string) *templatev1.Template {
```

Comment:
These two functions `getTemplateWithCurrentVersion()` and `getTemplateWithOldVersion()` can be a single function. Their implementation is the same except for template name.